### ISSUE TITLE
Update from 6.0.3.4 to 6.1.0.rc2  fails with a bootsnap error

### ISSUE THREAD

**Issue by Diogomartf** [rails/rails#40726]
Hey! Thanks for working hard on rails!

I was trying to update the app from 6.0.3.4 to 6.1.0.rc2 but it fails on Bootsnap ( I think ), after running `bundle update` I can't start the server or run `rails app:update`

### Steps to reproduce

1. Updated rails gem on the gemfile to 6.1.0.rc2 ( 6.1.0.rc1 has the same error )
2. bundle update
3. rails app:update  ( triggers the error )

Also running `rails server` or anything will trigger the error.

### Expected behavior
Running the server normally or running the command.

### Actual behavior
When running `rails app:update`
```ruby
rails aborted!
LoadError: cannot load such file -- thread_safe
/Users/diogo/code/gymious/gymious-rails-app/config/application.rb:7:in `<main>'
/Users/diogo/code/gymious/gymious-rails-app/Rakefile:4:in `<main>'
bin/rails:4:in `<main>'
(See full trace by running task with --trace)
```

When running `rails server`
```ruby
Traceback (most recent call last):
	56: from bin/rails:4:in `<main>'
	55: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/activesupport-6.1.0.rc2/lib/active_support/dependencies.rb:332:in `require'
	54: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/activesupport-6.1.0.rc2/lib/active_support/dependencies.rb:299:in `load_dependency'
	53: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/activesupport-6.1.0.rc2/lib/active_support/dependencies.rb:332:in `block in require'
	52: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
	51: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	50: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	49: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
	48: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
	47: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/railties-6.1.0.rc2/lib/rails/commands.rb:18:in `<main>'
	46: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/railties-6.1.0.rc2/lib/rails/command.rb:50:in `invoke'
	45: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/railties-6.1.0.rc2/lib/rails/command/base.rb:69:in `perform'
	44: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/thor-1.0.1/lib/thor.rb:392:in `dispatch'
	43: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/thor-1.0.1/lib/thor/invocation.rb:127:in `invoke_command'
	42: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/thor-1.0.1/lib/thor/command.rb:27:in `run'
	41: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/railties-6.1.0.rc2/lib/rails/commands/server/server_command.rb:135:in `perform'
	40: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/railties-6.1.0.rc2/lib/rails/commands/server/server_command.rb:135:in `tap'
	39: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/railties-6.1.0.rc2/lib/rails/commands/server/server_command.rb:138:in `block in perform'
	38: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/activesupport-6.1.0.rc2/lib/active_support/dependencies.rb:332:in `require'
	37: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/activesupport-6.1.0.rc2/lib/active_support/dependencies.rb:299:in `load_dependency'
	36: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/activesupport-6.1.0.rc2/lib/active_support/dependencies.rb:332:in `block in require'
	35: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
	34: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	33: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	32: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
	31: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
	30: from /Users/diogo/code/gymious/gymious-rails-app/config/application.rb:7:in `<main>'
	29: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/site_ruby/2.7.0/bundler.rb:174:in `require'
	28: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/site_ruby/2.7.0/bundler/runtime.rb:58:in `require'
	27: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/site_ruby/2.7.0/bundler/runtime.rb:58:in `each'
	26: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/site_ruby/2.7.0/bundler/runtime.rb:69:in `block in require'
	25: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/site_ruby/2.7.0/bundler/runtime.rb:69:in `each'
	24: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/site_ruby/2.7.0/bundler/runtime.rb:74:in `block (2 levels) in require'
	23: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
	22: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	21: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	20: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
	19: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
	18: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/active_model_serializers-0.10.10/lib/active_model_serializers.rb:8:in `<main>'
	17: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/active_model_serializers-0.10.10/lib/active_model_serializers.rb:60:in `<module:ActiveModelSerializers>'
	16: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/activesupport-6.1.0.rc2/lib/active_support/dependencies.rb:332:in `require'
	15: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/activesupport-6.1.0.rc2/lib/active_support/dependencies.rb:299:in `load_dependency'
	14: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/activesupport-6.1.0.rc2/lib/active_support/dependencies.rb:332:in `block in require'
	13: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
	12: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	11: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	10: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
	 9: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
	 8: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/active_model_serializers-0.10.10/lib/active_model/serializer.rb:3:in `<main>'
	 7: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/activesupport-6.1.0.rc2/lib/active_support/dependencies.rb:332:in `require'
	 6: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/activesupport-6.1.0.rc2/lib/active_support/dependencies.rb:299:in `load_dependency'
	 5: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/activesupport-6.1.0.rc2/lib/active_support/dependencies.rb:332:in `block in require'
	 4: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:44:in `require'
	 3: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
	 2: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/loaded_features_index.rb:89:in `register'
	 1: from /Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
/Users/diogo/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/bootsnap-1.5.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require': cannot load such file -- thread_safe (LoadError)
```

### System configuration
**Rails version**:
Rails 6.1.0.rc2
**Ruby version**:
ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-darwin19]

<details><summary>Gemfile</summary>

```ruby
source "https://rubygems.org"
git_source(:github) { |repo| "https://github.com/#{repo}.git" }

ruby "2.7.0"

gem "jsonapi-serializers"
# Random location
gem "random-location"

# JSON serialization
gem "active_model_serializers"
# Pagination
gem "kaminari"
# React
gem "react-rails"
# User authentication
gem "devise"
# Geo localization
gem "geocoder"
#images server
gem "cloudinary"
# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
gem "rails", "~> 6.1.0.rc2"
# Use pg as the database for Active Record
gem "pg"
# Use Puma as the app server
gem "puma", "~> 4.1"
# Use SCSS for stylesheets
gem "sass-rails", ">= 6"
# Transpile app-like JavaScript. Read more: https://github.com/rails/webpacker
gem "webpacker", "~> 4.0"
# Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks
gem "turbolinks", "~> 5"
# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
gem "jbuilder", "~> 2.7"
# Use Redis adapter to run Action Cable in production
# gem 'redis', '~> 4.0'
# Use Active Model has_secure_password
# gem 'bcrypt', '~> 3.1.7'

# Use Active Storage variant
# gem 'image_processing', '~> 1.2'

# Reduces boot times through caching; required in config/boot.rb
gem "bootsnap", ">= 1.4.2", require: false

group :development, :test do
  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
  gem "byebug", platforms: [:mri, :mingw, :x64_mingw]
  # Environment variables
  gem "dotenv-rails"
end

group :development do
  # Access an interactive console on exception pages or by calling 'console' anywhere in the code.
  gem "web-console", ">= 3.3.0"
  gem "listen", ">= 3.0.5", "< 3.2"
  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
  gem "spring"
  gem "spring-watcher-listen", "~> 2.0.0"
  # better errors page
  gem "better_errors"
  gem "binding_of_caller"
  #generate favicon
  gem "rails_real_favicon"
end

group :test do
  # Adds support for Capybara system testing and selenium driver
  gem "capybara", ">= 2.15"
  gem "selenium-webdriver"
  # Easy installation and use of web drivers to run system tests with browsers
  gem "webdrivers"
end

# Windows does not include zoneinfo files, so bundle the tzinfo-data gem
gem "tzinfo-data", platforms: [:mingw, :mswin, :x64_mingw, :jruby]

```
</details>

**Comment by jonathanhefner** [comment]
Thank you for opening an issue!  However, this error is from the `active_model_serializers` gem, rather than `bootsnap`.  From a quick search, it looks like it might be fixed by rails-api/active_model_serializers#2395, which has not been released yet (the last `active_model_serializers` release was [in 2019](https://rubygems.org/gems/active_model_serializers/versions)).  You might try the workaround mentioned in that thread (adding `thread_safe` to your Gemfile), or bundling the `0-10-stable` branch of the `active_model_serializers` repo.

**Comment by bf4** [comment]
ok, I just released active model serializers 0.10.11 with that fix.  (though I'd recommend you consider other gems at this point as it's in maintenance mode.  My last attempt to begin the third (?) rewrite of it stalled when I stopped working on the app I was extracting it from).
